src/presenter/point-presenter.js

```diff
@@ -0,0 +1,112 @@
+import { render, replace } from '../framework/render.js';
+import EventsView from '../view/event-view.js';
+import EventEdit from '../view/event-edit.js';
+
+const PointMode = {
+  VIEW: 'view',
+  EDIT: 'edit',
+};
+
+export default class PointPresenter{
+
+  #eventsListView = null;
+  #point = null;
+  #offer = null;
+  #destination = null;
+  #points = null;
+
+  #pointComponent = null;
+  #pointEditComponent = null;
+  #openEditModeHandler = null;
+
+  #mode = null;
+  #modeFavorit = null;
+  #openFavoritModeHandler = null;
+  #changeFavorite = null;
+
+  constructor({eventsListView, point, offers, destinations}){
+    this.#eventsListView = eventsListView;
+    this.#point = point;
+    this.#offer = offers;
+    this.#destination = destinations;
+    this.#mode = PointMode.VIEW;
+    this.#modeFavorit = this.#point.isFavorite;
+  }
+
+  #renderPoint = (point) => {
+    this.#pointComponent = new EventsView({
+      point,
+      pointDestination : this.#destination.getById(point.destination),
+      pointOffers: this.#offer.getByType(point.type)
+    });
+
+    this.#pointEditComponent = new EventEdit({
+      point,
+      pointDestination : this.#destination.get(),
+      pointOffers: this.#offer.get()
+    });
+
+    this.#pointEditComponent.setEditHandler(this.closeEditMode);
+    this.#pointComponent.setEditHandler(this.openEditMode);
+
+    this.#pointComponent.setFavoritHandler(this.changeFavorite);
+
+    render(this.#pointComponent,this.#eventsListView.element);
+  };
+
+
+  #replacePointToEdit = () => {
+    this.#mode = PointMode.EDIT;
+    replace(this.#pointEditComponent, this.#pointComponent);
+
+  };
+
+  #replaceEditToPoint = () => {
+    this.#mode = PointMode.VIEW;
+    replace(this.#pointComponent, this.#pointEditComponent);
+  };
+
+  openEditMode = () =>{
+    if(this.#mode === PointMode.VIEW){
+      this.#replacePointToEdit();
+      this.#openEditModeHandler?.();
+    }
+  };
+
+  setOpenEditModeHandler = (cb) =>{
+    this.#openEditModeHandler = cb;
+
+  };
+
+  closeEditMode = () => {
+    if(this.#mode === PointMode.EDIT){
+      this.#replaceEditToPoint();
+    }
+  };
+
+
+  setChangeFavoriteHandler = (cb) => {
+    this.#openFavoritModeHandler = cb;
+  };
+
+  setValueFavorite = () => {
+    this.#point.isFavorite = this.#modeFavorit;
+  };
+
+  changeFavorite = () => {
```

Comment:
Метод пересылающий изимененные данные в родительский контент: тут мы оставляем все данные кроме флага isFavorite неизменными. Т.е. мы создаем новый объект с данными точки, но меняем у неё одно значение и (!) отправляем эти данные наверх ( чтобы они дошли до модели, внутри неё мы эти данные заменим и сообщим об этом презентору родительскому, который сообщит нашему и обновим (заново отрисуем эту точку уже с нвоыми данными через init метод))
```suggestion
 changeFavorite = () => {
    this.#handleDataChange({
      ...this.#pointData,
      isFavorite: !this.#pointData.isFavorite
    });
  };
```